Scripts/Runtime/Entities/Lifecycle/EntityLifecycleStatus.cs

```diff
@@ -0,0 +1,251 @@
+using Anvil.CSharp.Core;
+using Anvil.Unity.DOTS.Jobs;
+using System.Linq;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    internal class EntityLifecycleStatus : AbstractAnvilBase,
+                                           IEntityLifecycleStatus
+    {
+        private readonly AccessControlledValue<NativeParallelHashSet<Entity>> m_Lookup;
+        private readonly AccessControlledValue<NativeList<Entity>> m_ArrivedEntities;
+        private readonly AccessControlledValue<NativeList<Entity>> m_DepartedEntities;
+        private readonly AbstractEntityLifecycleStatusSystem m_OwningSystem;
+        private readonly ComponentType[] m_QueryComponentTypes;
+
+        private EntityQuery m_Query;
+
+        public EntityLifecycleStatus(AbstractEntityLifecycleStatusSystem owningSystem, params ComponentType[] queryComponentTypes)
```

Comment:
Maybe we should accept `EntityQueryDesc` instead? Or at add support another day.

Reply:
I'm fine with adding it another day, I find I normally just like using the params array for faster typing and not needing to create the desc object

Reply:
Yea, maybe an overload then. I could see wanting to submit a more advanced query that used things like exclusions.

Reply:
Yep that's fair, although you can do `ComponentType.Exclude<T>`

Added the TODO